os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -183,30 +155,25 @@ void binary_manager_register_kpart(int part_num, int part_size)
 bool binary_manager_scan_kbin(void)
 {
 	int ret;
-	int part_idx;
-	uint32_t latest_ver;
+	binmgr_bpdata_t *bp_data;
 	kernel_binary_header_t header_data;
 	char filepath[BINARY_PATH_LEN];
 
-	latest_ver = 0;
-
-	/* Load the binaries with high priority directly */
-	for (part_idx = 0; part_idx < kernel_info.part_count; part_idx++) {
-		snprintf(filepath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, kernel_info.part_num[part_idx]);
+	ret = binary_manager_scan_bootparam();
+	if (ret == BINMGR_OK) {
+		bp_data = binary_manager_get_bpdata();
+		/* Verify running kernel binary based on bootparam */
+		snprintf(filepath, BINARY_PATH_LEN, BINMGR_DEVNAME_FMT, kernel_info.part_num[bp_data->active_idx]);
 		ret = binary_manager_verify_kernel_binary(filepath, &header_data);
-		if (ret == OK && latest_ver < header_data.version) {
-			/* Update latest version and inuse index */
+		if (ret == OK) {
+			/* Update inuse index and kernel version */
+			kernel_info.inuse_idx = bp_data->active_idx;
 			kernel_info.version = header_data.version;
-			kernel_info.inuse_idx = part_idx;
-			latest_ver = header_data.version;
+			bmvdbg("Kernel version [%u] %u\n", kernel_info.inuse_idx, kernel_info.version);
+			return true;
 		}
 	}
-	bmvdbg("Latest version [%u] %u\n", kernel_info.inuse_idx, kernel_info.version);
 
-	/* Found valid binary */
-	if (kernel_info.version != 0) {
-		return true;
-	}
 	return false;
```

Comment:
Even if it returns false because of failure to scan the boot param, the log message says it is from parsing partition list.
It is not good to debug when fail.
Please distinguish the log message for failure case.

Reply:
I think it is enough because each cases of failure print error logs.

Reply:
I mean, even the failure is from bootparam, but the error logs says it is from parsing of partition list.
```
		if (!is_found_kpart) {
			lldbg("ERROR!! Not found kernel partitions because parsing a partition list is failed.\n");
			lldbg("Please check logs from configure_mtd_partitions and whether the partition 'kernel' exists in CONFIG_FLASH_PART_TYPE.\n");
		}
```
Do you think it is enough?

Reply:
I added logs for failures of bootparam scanning.